acasclient/lsthing.py

```diff
@@ -2170,12 +2174,14 @@ def __init__(self, verb=None, subject=None, object=None, metadata=None, results=
                 if subject:
                     first_ls_thing = subject._ls_thing
                     first_type = subject.ls_type
-                else:
+                # If subject_type is provided, use it instead of the subject's ls_type
+                if subject_type:
```

Comment:
We use subject_type if it is passed in as a param instead of only if the subject is not passed in.  This gives more explicit control to the user of SimpleLink to specify the subject_type.  This applies to the object_type as well below.